### PR TITLE
Release 0.20.0

### PR DESCRIPTION
**Pull request by tatiana**
### Added
- Support using envvar in config YAML by @tatiana in #236
- **Callback improvements**
  - Support installed code via python callable string by @john-drews in #221
  - Add `callback_file` & `callback_name` to `default_args` DAG level by @subbota19 in #218
  - Cast callbacks to functions when set with `default_args` on TaskGroups by @Baraldo and @pankajastro in #235

- **Telemetry**
  - For more information, please, read the [Privacy Notice](https://github.com/astronomer/dag-factory/blob/main/PRIVACY_NOTICE.md#collection-of-data).
  - Add scarf to readme for website analytics by @cmarteepants in #219
  - Support telemetry during DAG parsing emitting data to Scarf by @tatiana in #250.

### Fixed
- Build DAGs when tehre is an invalid YAML in the DAGs folder by @quydx and @tatiana in #184

### Others
- Development tools
  - Fix make docker-run by @pankajkoti in #249
  - Add vim dot files to .gitignore by @tatiana in #228
  - Use Hatchling to modern package building by @kaxil in #208
- CI
  - Fix static check failures in PR #218 by @pankajkoti in #251
  - Fix pre-commit checks by @tatiana in #247
  - Remove tox and corresponding build jobs in CI by @pankajkoti in #248
  - Install Airflow with different versions in the CI by @pankajkoti in #237
  - Run pre-commit hooks on all existing files by @pankajkoti in #245
  - Add Python 3.11 and 3.12 to CI test pipeline by @pankajkoti in #229
- Tests
  - Fix duplicate test name by @pankajastro in #234
  - Add static check by @pankajastro in #231
  - Fix running tests locally (outside the CI) by @tatiana in #227
  - Add the task_2 back to dataset example by @cmarteepants in #204
  - Remove unnecessary config line by @jlaneve in #202
- Documentation
  - Update the license from MIT to Apache 2.0 by @pankajastro in #191
  - Add registration icon and links to Airflow references by @cmarteepants in #190
  - Update quickstart and add feature examples by @cmarteepants #189

### Breaking changes
- Removed support for Python 3.7
- The license was changed from MIT to Apache 2.0

Closes: #217

### CODE REVIEW

**Comment by tatiana** [comment]
We just released an alpha version, `0.20.0a1`, with the changes in the description of the PR, as of d67be8d3eb25b1c61eaaf2dd609d9044ce40c9f2.

Please, @john-drews @subbota19 @Baraldo @quydx, if you could support us validating this DAG Factory pre-release, it would be very helpful:
https://pypi.org/project/dag-factory/0.20.0a1/

**Comment by Baraldo** [comment]
Hi there @tatiana, for our use case it's all set!

File: dagfactory/__init__.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 from .dagfactory import DagFactory, load_yaml_dags
 
-__version__ = "0.20.0a2"
+__version__ = "0.20.0rc1"
```
Comment (pankajastro) [inline dagfactory/__init__.py:5]:
```suggestion
__version__ = "0.20.0"
```

**Review by pankajastro — APPROVED** [review]
LGTM

File: dagfactory/__init__.py (hunk 1 above)
Comment (pankajkoti) [inline dagfactory/__init__.py:5]:
```suggestion
__version__ = "0.20.0"
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,8 +5,66 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
+  - Add scarf to readme for website analytics by @cmarteepants in #219
+  - Support telemetry during DAG parsing emitting data to Scarf by @tatiana in #250.
+
+### Fixed
+- Build DAGs when there is an invalid YAML in the DAGs folder by @quydx and @tatiana in #184
+
+### Others
+- Development tools
+  - Fix make docker-run by @pankajkoti in #249
+  - Add vim dot files to .gitignore by @tatiana in #228
```
Comment (pankajkoti) [inline CHANGELOG.md:29]:
```suggestion
  - Add vim dot files to .gitignore by @tatiana in #228
  - Add local airflow setup files to .gitignore by @pankajkoti in #246
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,8 +5,66 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
+  - Run integration tests in CI by @pankajkoti in #266
+  - Improve test coverage by @pankajastro in #258
+- Refactor
+  - Refactor poor exception handling by @tatiana in #259
+  - Remove off looking start-date value in example_dag yaml config by @pankajkoti in #273
+- Documentation
+  - Update the license from MIT to Apache 2.0 by @pankajastro in #191
+  - Add registration icon and links to Airflow references by @cmarteepants in #190
+  - Update quickstart and add feature examples by @cmarteepants #189
+  - Fix `README.md` badges by @tatiana in #260
```
Comment (pankajkoti) [inline CHANGELOG.md:62]:
```suggestion
  - Fix `README.md` badges by @tatiana in #260
  - Remove duplicated operator in `README.md` by @brair in #263 
```

**Review by pankajkoti — APPROVED** [review]
Looks good to me. Thanks for drafting this release! I’ve left a couple of inline suggestions that caught my attention, which we might want to include in the Changelog as well.

**Pull request merged**